src/App.js

```diff
@@ -1,26 +1,120 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import Main from './components/Main';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      items: [],
+      selectedList: 0,
+      checked: false,
+    };
+
+    this.addToValue = (event) => {
+      this.setState({
+        value: event.target.value,
+      });
+    };
+
+    this.addToItems = () => {
+      if (this.state.value) {
+        this.setState(prevState => ({
+          items: [
+            ...prevState.items,
+            {
+              value: this.state.value,
+              done: false,
+            },
+          ],
+
+          value: '',
+        }));
+      }
+    };
+
+    this.handleClickToDone = (value) => {
+      for (const key of this.state.items) {
+        if (key.value === value) {
+          key.done = !key.done;
+        }
+      }
+
+      this.setState({
+        items: this.state.items,
+
+      });
+    };
+
+    this.showAllItems = () => {
+      this.setState({ selectedList: 0 });
+    };
+
+    this.showActiveItems = () => {
+      this.setState({ selectedList: 1 });
+    };
+
+    this.showCompletedItems = () => {
+      this.setState({ selectedList: 2 });
+    };
+
+    this.removeAllCompletedItems = () => {
+      const items = this.state.items.filter(item => !item.done);
+      this.setState({ items });
+    };
+
+    this.removeCompletedItem = (value) => {
+      const newValue = this.state.items.filter(item => item.value !== value);
+
+      this.setState({ items: newValue });
+    };
+
+    this.checkedAllItems = () => {
+      for (const key of this.state.items) {
+        key.done = !this.state.checked;
```

Comment:
The same as above. Don't mutate state directly. You are supposed to use only `setState`.
Suggestion:
```javascript
const items = this.state.items.map((item) => ({
  ...item,
  done: !this.state.checked,
}))

this.setState({
  items,
  checked: !this.state.checked
})
```